test/functional/pdo_sqlsrv/pdo_connect_driver.phpt

```diff
@@ -0,0 +1,186 @@
+--TEST--
+Test new connection keyword Driver with valid and invalid values
+--SKIPIF--
+<?php require('skipif.inc'); ?>
+--FILE--
+<?php
+require_once( 'MsSetup.inc' );
+
+try
+{
+    $conn = new PDO( "sqlsrv:server = $server", $uid, $pwd );
+    $msodbcsql_ver = $conn->getAttribute( PDO::ATTR_CLIENT_VERSION )['DriverVer'];
+    $msodbcsql_maj = explode(".", $msodbcsql_ver)[0];
+}
+catch( PDOException $e )
+{
+    echo "Failed to connect\n";
+    print_r( $e->getMessage() );
+    echo "\n";
+}
+
+$conn = null;
+
+// start test
+test_valid_values();
+test_invalid_values();
+test_encrypted_with_odbc();
+test_wrong_odbc();
+echo "Done";
+// end test
+
+///////////////////////////
+function connect_verify_output( $connectionOptions, $expected = '' )
+{
+    global $server, $uid, $pwd;
+    
+    try
+    {
+        $conn = new PDO( "sqlsrv:server = $server ; $connectionOptions", $uid, $pwd );
+    }
+    catch( PDOException $e )
+    {
+        if ( strpos($e->getMessage(), $expected ) === false )
+        {
+            print_r( $e->getMessage() );
+            echo "\n";
+        }
+    }
+}
+
+function test_valid_values()
+{
+    global $msodbcsql_maj;
+    
+    $value = "";
+    // Test with {}
+    switch ( $msodbcsql_maj )
+    {
+        case 17:
+            $value = "{ODBC Driver 17 for SQL Server}";
+            break;
+        case 13:
+            $value = "{ODBC Driver 13 for SQL Server}";
+            break;
+        case 12:
```

Comment:
there is no 12

Reply:
newer versions of ODBC 11 do return 12

Reply:
you are right! pls add a comment about this